src/python/grpcio_tests/tests/testing/_client_test.py

```diff
@@ -0,0 +1,316 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+import grpc
+from grpc.framework.foundation import logging_pool
+from tests.unit.framework.common import test_constants
+import grpc_testing
+
+from tests.testing import _application_common
+from tests.testing import _application_testing_common
+from tests.testing import _client_application
+
+
+class ClientTest(unittest.TestCase):
+
+    def setUp(self):
+        self._application_thread_pool = logging_pool.pool(1)
```

Comment:
Interesting. My default design in this situation is to have the client-side run in the foreground, with the mock server behavior happening in the background. Now everything makes sense.

Maybe `client_execution_thread_pool`?

Maybe a comment that the server is running in the foreground here? I'm prepared to believe I'm the exception vis-a-vis where mocks run.